lsp-mode.el

```diff
@@ -6195,9 +6195,11 @@ information, for example if it doesn't support DocumentSymbols."
   "Render INPUT0, an `&DocumentSymbol', to a string.
 If SHOW-DETAIL? is set, make use of its `:detail?' field (often
 the signature)."
-  (let ((base (or (and show-detail? detail?) detail? name)))
-    (if deprecated? (propertize base 'face 'lsp-face-semhl-deprecated)
-      base)))
+  (let ((detail (and show-detail? (s-present? detail?)
+                     (propertize (concat ": " detail?) 'face 'font-lock-type-face)))
```

Comment:
": " is not needed

Reply:
Compare without and with colon:
![20201107-102701](https://user-images.githubusercontent.com/69264154/98437615-dc435f80-20e3-11eb-972b-cf3b8a4f658c.png)
![20201107-102640](https://user-images.githubusercontent.com/69264154/98437620-e1a0aa00-20e3-11eb-99f6-6092985f7275.png)

Are you sure? I think I prefer the colon version, because it more clearly separates signature from name (and VSCode does it too), but I am of course open to different opinions.

Reply:
Here it is the JDTLS detail:

>  "detail": " : void"

so adding ":" is not on the table.

Reply:
True, that would look awkward. However, we cannot get rid of the double space, as everything else would look strange
![20201107-103447](https://user-images.githubusercontent.com/69264154/98437729-ee71cd80-20e4-11eb-8be2-c0c413b5f407.png)
 then:

Reply:
Perhaps we should fix this upstream?

Reply:
I guess we have to trim the detail

Reply:
> Perhaps we should fix this upstream?

upstream does not care as long as it works with vscode.

Reply:
Does VSCode also do the trimming?

Reply:
From looking at the VSCode outline, it doesn't seem to trim, either? But I guess we'll have to fold, as fixing this upstream might be a pain.

Reply:
Seems like they are not trimming. Feel free to trim/not to trim. IMO both are acceptable.

Looking on imenu-list on your screenshot seems like the detail does not look very well(at least it looks very noisy/unreadable due to using the same font). What do you think about disabling detail for imenu by default?

Reply:
Actually, I think it looks fine. Here is another screenshot with `lsp-treemacs-symbols`, `imenu` and `imenu-list`:
![20201107-105209](https://user-images.githubusercontent.com/69264154/98438010-5de8bc80-20e7-11eb-982e-6dbfd34db43a.png)
For some reason, imenu entries get cut off, but that isn't an issue with lsp-mode.

Reply:
`imenu` might be too dark though.